DataBiosphere/tanagra
# Add ConceptSet.clone

- Adds support for Clone Concept Set (FeatureSet)
- https://github.com/DataBiosphere/tanagra/pull/972 renames this to FeatureSet. Coordinate with this PR for merge
- Update Concept Set test
- Create TestUtils with user email as static strings and use them in other tests
- https://verily.atlassian.net/browse/BENCH-3850
----
Manual test

```
Create study
Create concept set, verify
Updated concept set 
{
  "id": "COWxMkSumI",
  "underlayName": "cmssynpuf",
  "displayName": "cs-2",
  "description": "cs-2",
  "criteria": [
    {
      "id": "yISxOFco8F",
      "displayName": "c-1",
      "pluginName": "test pluginName",
      "pluginVersion": 0,
      "predefinedId": "string",
      "selectorOrModifierName": "string",
      "selectionData": "string",
      "uiConfig": "ui",
      "tags": {
        "k1": "v1"
      }
    }
  ],
  "entityOutputs": [
    {
      "entity": "person",
      "excludeAttributes": [
        "age"
      ]
    }
  ],
  "created": "2024-09-02T01:34:41.49677Z",
  "createdBy": "authentication-disabled",
  "lastModified": "2024-09-02T08:39:50.174887Z"
}

Clone:

curl -X 'POST' \
  'http://localhost:8080/v2/studies/study-1/conceptSets/COWxMkSumI/clone' \
  -H 'accept: */*' \
  -H 'Content-Type: application/json' \
  -d '{
  "displayName": "cloned cs",
  "description": "cloned cs desc"
}'


{
  "id": "cBPS4IaCo4",
  "underlayName": "cmssynpuf",
  "displayName": "cloned cs",
  "description": "cloned cs desc",
  "criteria": [
    {
      "id": "yISxOFco8F",
      "displayName": "c-1",
      "pluginName": "test pluginName",
      "pluginVersion": 0,
      "predefinedId": "string",
      "selectorOrModifierName": "string",
      "selectionData": "string",
      "uiConfig": "ui",
      "tags": {
        "k1": "v1"
      }
    }
  ],
  "entityOutputs": [
    {
      "entity": "person",
      "excludeAttributes": [
        "age"
      ]
    }
  ],
  "created": "2024-09-02T01:41:17.963329Z",
  "createdBy": "authentication-disabled",
  "lastModified": "2024-09-02T01:41:17.963329Z"
}

Verified that both concepts are displayed with list
```
